### PR TITLE
add Vault.sol and Authorize.sol

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

**Review by mod — CHANGES_REQUESTED** [review]
On my side I would:
- [x] Switch RBAC to onlyOwner as we have only 1 role
- [x] Avoid using enumarableMap as you don't use it
- [ ] Probably we can remove OZ fully, like copying IERC20.sol

File: src/TimeRangeAuthorizer.sol [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/IAuthorize.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
```
Comment (mod) [inline src/TimeRangeAuthorizer.sol]: use Owner RBAC if no multirole

**Comment by mnaichuk** [comment]
> On my side I would:
> 
> * [x]  Switch RBAC to onlyOwner as we have only 1 role
> * [x]  Avoid using enumarableMap as you don't use it
> * [ ]  Probably we can remove OZ fully, like copying IERC20.sol

I would still keep ReentrancyGuard from OZ

**Review by mod — CHANGES_REQUESTED** [review]
- [x] Comments
- [x] Align to TimeRange naming
- [x] Rename Vault to LiteVault

File: src/LiteVault.sol [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    event Deposit(address indexed user, address indexed token, uint256 amount);
+    event Withdrawal(address indexed user, address indexed token, uint256 amount);
+
+    /**
+     * @dev Modifier to check if the caller is the owner of the contract.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not the contract owner");
+        _;
+    }
```
Comment (mod) [inline src/LiteVault.sol:29]: Todo, we need to add the ability to transfer Ownership,
Either we copy Ownable.sol or implement here

**Review by mod — APPROVED** [review]

**Pull request merged**